### ISSUE TITLE
Allow Users to have custom themes

### ISSUE THREAD

**Issue by JakobDev** [go-gitea/gitea#23524]
### Feature Description

At the moment, it's only possible to add themes for a Gitea installation. If you are just a User and not the Admin, you can't add a Theme. The only Way to use a custom theme is asking the Admin, if he could add it or by using a Extension like Stylus. I think we should add the possibility for Users to set a custom theme. Maybe by adding a link to a CSS in the Settings.

**Comment by silverwind** [comment]
Maybe just offer a box for a user to paste Theme CSS which is a lower barrier of entry than requiring external hosting.

That said, theming is still not quite where I like it to be (CSS Vars only), we still need to variablize the chroma and codemirror colors.

**Comment by Mosch0512** [comment]
This would be great.

**Comment by mcepl** [comment]
The design of the HTML is so convoluted, that I have not been even able to create a good `userContent.css` to distinguish between our community Gitea and the internal one.

**Comment by lunny** [comment]
You can add a custom logo to make it easier to distinguish between the different instances.

**Comment by mcepl** [comment]
> You can add a custom logo to make it easier to distinguish between the different instances.

Yes, but even with one line `userContent.css` I could make something which works for me even without thinking (not that it is the most stylish thing ever, I don’t claim that, but it surely works), see screenshots in https://github.com/openSUSE/openSUSE-git/issues/186
